665053: snmpd stucks when processing smux MIB on big endian systems
720704 - Assigning negative number to unsigned variable

Source: upstream, SVN rev. 19108, git commit 40125d8539cd89f952fbc24588c5e81e9cd821fe

--- net-snmp-5.5/agent/mibgroup/smux/smux.c.smux	2010-12-22 15:01:48.708105735 +0100
+++ net-snmp-5.5/agent/mibgroup/smux/smux.c	2010-12-22 15:01:02.000000000 +0100
@@ -1356,7 +1356,7 @@ smux_snmp_process(int exact,
                   size_t * return_len, u_char * return_type, int sd)
 {
     u_char          packet[SMUXMAXPKTSIZE], *ptr, result[SMUXMAXPKTSIZE];
-    int             length = SMUXMAXPKTSIZE;
+    ssize_t         length = SMUXMAXPKTSIZE;
     int             tmp_length;
     u_char          type;
     size_t          packet_len;
